pkg/kubectl/service.go

```diff
@@ -134,10 +147,14 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 		if len(portStringSlice) > 1 {
 			name = fmt.Sprintf("port-%d", i+1)
 		}
+		protocol := params["protocol"]
```

Comment:
What should happen if I expose a service that has multiple ports with different protocols and also use --protocol? It seems that my value will be overriden, right? This is a problem with setting a default in the protocol flag.